### PR TITLE
Add tools for dynamic simulations

### PR DESCRIPTION
**Pull request by flo-dup**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**
Feature

**What is the new behavior (if this is a feature change)?**
Following artifacts have been added:
- powsybl-dynamic-simulation-tool
- powsybl-action-dsl
- powsybl-action-dsl-spi
- powsybl-action-simulator
- powsybl-action-util

**Does this PR introduce a breaking change or deprecate an API?**
No

### CODE REVIEW

**Comment by flo-dup** [comment]
Shouldn't we also add `powsybl-starter` artifact itself?

**Comment by geofjamg** [comment]
> Shouldn't we also add `powsybl-starter` artifact itself?

This is not the purpose of powsybl-starter.

File: pom.xml [hunk 1]
```diff
@@ -189,6 +189,36 @@
             <scope>runtime</scope>
         </dependency>
 
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-dynamic-simulation-tool</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+
+        <dependency>
```
Comment (geofjamg) [inline pom.xml:198]: Why do we need to add actions related artifacts? It has nothing to do with dynamic simulation?

Comment (flo-dup) [reply pom.xml:198]: Removing them. They were added by mistake, in order to add `GroovyDslContingenciesProviderFactory` which was spotted in `powsybl-action-dsl`. But in fact that factory is already in `powsybl-contingency-dsl`.

**Review by geofjamg — APPROVED** [review]

**Pull request merged**